curator-x-async/src/test/java/org/apache/curator/x/async/modeled/TestCachedModeledFramework.java

```diff
@@ -181,6 +184,72 @@ public void testAccessCacheDirectly()
         }
     }
 
+    // Verify the CachedModeledFramework does not attempt to deserialize empty ZNodes on deletion.
+    // See: CURATOR-609
+    @Test
+    public void testEmptyNodeDeserialization()
+    {
+        // The sub-path is the ZNode that will be removed that does not contain a model. This model with no data
+        // should not result in attempt to deserialize.
+        final String subPath = modelSpec.path().toString() + "/sub";
+
+        final String firstModelPath = subPath + "/first";
+        final String secondModelPath = subPath + "/second";
+
+        CountDownLatch latch = new CountDownLatch(1);
+
+        final AtomicBoolean caughtException = new AtomicBoolean(false);
+
+        // Create a custom listener to signal the end of the test and ensure no exceptions are thrown.
+        final ModeledCacheListener<TestModel> listener = new ModeledCacheListener<TestModel>() {
+            @Override
+            public void accept(Type t, ZPath p, Stat s, TestModel m)
+            {
+                if (t == ModeledCacheListener.Type.NODE_ADDED && p.toString().equals(secondModelPath)) {
+                    latch.countDown();
+                }
+            }
+
+            public void handleException(Exception e) {
+                System.err.printf("Caught unexpected exception %s%n", e);
```

Comment:
This is unnecessary. Or you could try `AtomicReference<Exception>` and `assertNull` ? This will give us cue about failure reason without resort to `printf`.

Reply:
I like storing the Exception.  I tried to assertFail right there, but the exception is swallowed somewhere in the implementation and didn't cause the test to fail.

Reply:
We are failing what others failed. I experienced this too.